src/main/kotlin/uk/gov/gdx/datashare/services/SuppliersService.kt

```diff
@@ -124,4 +124,27 @@ class SuppliersService(
       )
     }
   }
+
+  fun deleteSupplier(
+    id: UUID,
+  ): Supplier {
+    val now = dateTimeHandler.now()
+    adminActionAlertsService.noticeAction(
+      AdminAction(
+        "Delete supplier",
+        object {
+          val supplierId = id
+          val whenDeleted = now
+        },
+      ),
+    )
+    val supplier = supplierRepository.save(
+      supplierRepository.findByIdOrNull(id)?.copy(
+        whenDeleted = now,
+      ) ?: throw SupplierSubscriptionNotFoundException("Supplier $id not found"),
```

Comment:
I'm not convinced this should be an error - if you call delete twice should it return OK because the entity has been deleted?